graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java

```diff
@@ -470,30 +477,62 @@ public EntityExcerpt createExcerpt(InputWithExtractors inputWithExtractors) {
         try {
             final Input input = inputService.find(modelId.toString());
             final InputWithExtractors inputWithExtractors = InputWithExtractors.create(input, inputService.getExtractors(input));
-            final Stream<String> extractorLookupNames = inputWithExtractors.extractors().stream()
-                    .filter(e -> e.getType().equals(Extractor.Type.LOOKUP_TABLE))
-                    .map(e -> (String) e.getExtractorConfig().get(LookupTableExtractor.CONFIG_LUT_NAME));
-            final Stream<String> converterLookupNames = inputWithExtractors.extractors().stream()
-                    .flatMap(e -> e.getConverters().stream())
-                    .filter(c -> c.getType().equals(Converter.Type.LOOKUP_TABLE))
-                    .map(c -> (String) c.getConfig().get("lookup_table_name"));
-
-            Stream.concat(extractorLookupNames, converterLookupNames)
-                    .map(lookupTableService::get)
-                    .filter(Optional::isPresent)
-                    .map(Optional::get)
-                    .forEach(lookupTableDto -> {
-                        EntityDescriptor lookupTable = EntityDescriptor.create(
-                                ModelId.of(lookupTableDto.id()), ModelTypes.LOOKUP_TABLE_V1);
-                        mutableGraph.putEdge(entityDescriptor, lookupTable);
-                    });
 
+            final MutableGraph<EntityDescriptor> lookupGraph = resolveNativeEntityLookupTable(entityDescriptor,
+                    inputWithExtractors, mutableGraph);
+            final MutableGraph<EntityDescriptor> grokGraph = resolveNativeEntityGrokPattern(entityDescriptor,
+                    inputWithExtractors, lookupGraph);
```

Comment:
What do you think about just passing the `mutableGraph` to both of these methods instead of passing the result of one into the other one? That makes it dependent on the order of method calls.

Since the graph is mutable anyway, we can just pass it to the methods and add the edges directly.

```diff
diff --git graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
index a58ba21a8..fff03c324 100644
--- graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
+++ graylog2-server/src/main/java/org/graylog2/contentpacks/facades/InputFacade.java
@@ -478,19 +478,17 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
             final Input input = inputService.find(modelId.toString());
             final InputWithExtractors inputWithExtractors = InputWithExtractors.create(input, inputService.getExtractors(input));
 
-            final MutableGraph<EntityDescriptor> lookupGraph = resolveNativeEntityLookupTable(entityDescriptor,
-                    inputWithExtractors, mutableGraph);
-            final MutableGraph<EntityDescriptor> grokGraph = resolveNativeEntityGrokPattern(entityDescriptor,
-                    inputWithExtractors, lookupGraph);
+            resolveNativeEntityLookupTable(entityDescriptor, inputWithExtractors, mutableGraph);
+            resolveNativeEntityGrokPattern(entityDescriptor, inputWithExtractors, mutableGraph);
 
-            return ImmutableGraph.copyOf(grokGraph);
+            return ImmutableGraph.copyOf(mutableGraph);
         } catch (NotFoundException e) {
             LOG.debug("Couldn't find input {}", entityDescriptor, e);
         }
         return ImmutableGraph.copyOf(mutableGraph);
     }
 
-    private MutableGraph<EntityDescriptor> resolveNativeEntityGrokPattern(EntityDescriptor entityDescriptor,
+    private void resolveNativeEntityGrokPattern(EntityDescriptor entityDescriptor,
                                                 InputWithExtractors inputWithExtractors,
                                                 MutableGraph<EntityDescriptor> mutableGraph) {
         inputWithExtractors.extractors().stream()
@@ -505,11 +503,9 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
                         mutableGraph.putEdge(entityDescriptor, depEntityDescriptor);
                     });
                 });
-
-        return mutableGraph;
     }
 
-    private MutableGraph<EntityDescriptor> resolveNativeEntityLookupTable(EntityDescriptor entityDescriptor,
+    private void resolveNativeEntityLookupTable(EntityDescriptor entityDescriptor,
                                                 InputWithExtractors inputWithExtractors,
                                                 MutableGraph<EntityDescriptor> mutableGraph) {
 
@@ -530,7 +526,6 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
                             ModelId.of(lookupTableDto.id()), ModelTypes.LOOKUP_TABLE_V1);
                     mutableGraph.putEdge(entityDescriptor, lookupTable);
                 });
-        return mutableGraph;
     }
 
     @Override
@@ -552,15 +547,13 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
 
         final InputEntity input = objectMapper.convertValue(entity.data(), InputEntity.class);
 
-        final MutableGraph<Entity> lookupGraph = resolveForInstallationV1LookupTable(entity,
-                input, parameters, entities, graph);
-        final MutableGraph<Entity> grokPatternGraph = resolveForInstallationV1GrokPattern(entity,
-                input, parameters, entities, lookupGraph);
+        resolveForInstallationV1LookupTable(entity, input, parameters, entities, graph);
+        resolveForInstallationV1GrokPattern(entity, input, parameters, entities, graph);
 
-        return ImmutableGraph.copyOf(grokPatternGraph);
+        return ImmutableGraph.copyOf(graph);
     }
 
-    private MutableGraph<Entity> resolveForInstallationV1GrokPattern(EntityV1 entity,
+    private void resolveForInstallationV1GrokPattern(EntityV1 entity,
                                                 InputEntity input,
                                                 Map<String, ValueReference> parameters,
                                                 Map<EntityDescriptor, Entity> entities,
@@ -581,10 +574,9 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
                                 return grokPatternEntity1.name().asString(parameters).equals(patternName);
                             }).forEach(x -> graph.putEdge(entity, x.getValue()));
                 });
-        return graph;
     }
 
-    private MutableGraph<Entity> resolveForInstallationV1LookupTable(EntityV1 entity,
+    private void resolveForInstallationV1LookupTable(EntityV1 entity,
                                                      InputEntity input,
                                                      Map<String, ValueReference> parameters,
                                                      Map<EntityDescriptor, Entity> entities,
@@ -608,7 +600,5 @@ public class InputFacade implements EntityFacade<InputWithExtractors> {
                     return  lookupTableNames.contains(lookupTableEntity.name().asString(parameters));
                 })
                 .forEach(x -> graph.putEdge(entity, x.getValue()));
-
-        return graph;
     }
 }
```